jerry-core/mem/mem-heap.c

```diff
@@ -199,23 +146,29 @@ static void mem_heap_stat_free_iter ();
 void
 mem_heap_init (void)
 {
-  JERRY_STATIC_ASSERT ((uintptr_t) mem_heap.area % MEM_ALIGNMENT == 0,
+  JERRY_STATIC_ASSERT ((uintptr_t) MEM_HEAP_CONTEXT (area) % MEM_ALIGNMENT == 0,
```

Comment:
Careful with the static asserts. The `MEM_HEAP_CONTEXT` macro could change so it would not resolve statically. Either change it to a runtime assert or add a comment near `MEM_HEAP_CONTEXT` that states something like "this needs to resolve statically as it is used in static asserts"